tests/utils.py

```diff
@@ -93,13 +95,13 @@ def tearDown(self):
 
         # Check that the test data has not been modified, only catches changes in attack.generate if self has been used
         np.testing.assert_array_almost_equal(
-            self._x_train_mnist_original[0 : self.n_train], self.x_train_mnist, decimal=3
+            self._x_train_mnist_original[0: self.n_train], self.x_train_mnist, decimal=3
         )
         np.testing.assert_array_almost_equal(
-            self._y_train_mnist_original[0 : self.n_train], self.y_train_mnist, decimal=3
+            self._y_train_mnist_original[0: self.n_train], self.y_train_mnist, decimal=3
```

Comment:
Please don't remove these characters, this is following Black style.
```suggestion
            self._y_train_mnist_original[0 : self.n_train], self.y_train_mnist, decimal=3
```